### PR TITLE
Baby steps toward Geant4 integration

### PR DESCRIPTION
**Pull request by sethrj**
This is a collection of small changes motivated by Acceleritas integration.
- I moved a little code out of the demo loop IO setup: the geo-material coupling now can be set through the imported data.
- I made a few thread-friendly changes
- The geant4 importer tests now pass when running v11
- EM parameters are now a struct
- `GeantSetup` uses `RunInitialization` instead of `BeamOn`.
- Physics option names are now consistent between input and output

### CODE REVIEW

**Comment by sethrj** [comment]
@whokion I didn't make as many changes as I thought I would because I'm not quite sure of the right place to put Geant4-derived components. All the stuff in `celeritas/ext` is designed basically as "shim" classes to external libraries (i.e. you can include them whether or not the library is enabled), whereas for "run Celeritas from Geant4" the top-level classes are going to necessarily require Geant4, and we should probably not even build that library if geant4 isn't enabled. So perhaps we do want to end up with an `acceleritas` library under `src/`: and I want to wait until #526 is merged to start that.

File: src/celeritas/io/ImportData.hh
Comment (stognini) [inline src/celeritas/io/ImportData.hh:26]: Suggestion to update comment, since it's not an enum anymore.

File: app/celer-dump-data.cc
Comment (stognini) [inline app/celer-dump-data.cc:352]: Table looks off when printed. Not sure why, since the macro looks right at first sight...

File: test/celeritas/ext/GeantImporter.test.cc
Comment (stognini) [inline test/celeritas/ext/GeantImporter.test.cc:348]: Nice. We have a few spots that would break with different G4 data versions (which is correlated to the G4 version, unless manually configured). Makes me think if we want to make tests compatible with more than 1 data version...

File: app/demo-loop/LDemoIO.cc
Comment (stognini) [inline app/demo-loop/LDemoIO.cc:235]: Much nicer! Remapping internally also makes more sense. I see you removed the warning, is there any chance that we might miss having the message in case the label remapping ends up being the problem?

**Comment by stognini** [comment]
Looks great. I've got a couple of test failures on G4 `11.0.0`, but everything passes on `10.7.3` and `11.0.1` (?)... so I'll just double check if we missed anything on `11.0.0` or if it is my build that's the culprit. Other than, great work and I'm really glad we found out about the `RunInitialization()`. Having the full user action and a dummy primary always looked silly to me.

File: app/celer-dump-data.cc
Comment (sethrj) [reply app/celer-dump-data.cc:352]: This is a weird one to crack... looks like a combination of std::boolalpha and std::setw causing the `" |\n"` to be right-aligned and indented...

Comment (sethrj) [reply app/celer-dump-data.cc:352]: Ugh. https://timsong-cpp.github.io/lwg-issues/2703

Comment (sethrj) [reply app/celer-dump-data.cc:352]: So
```
   cout << std::boolalpha << "| " << setw(18) << false << "|\n";
   cout << std::noboolalpha << "| " << setw(18) << false <<  "|\n";
```
prints
```
| false|
                | 0                 |
```

File: app/demo-loop/LDemoIO.cc
Comment (sethrj) [reply app/demo-loop/LDemoIO.cc:235]: I don't think we'll miss the warning: in fact, knowing what I know now, it's only a "coincidence" when the names and IDs match up, because Geant4 and VecGeom build their models independently.

**Comment by sethrj** [comment]
@stognini For the sake of completion, could you post the 11.0.0 failures?

File: app/demo-loop/LDemoIO.cc
Comment (stognini) [reply app/demo-loop/LDemoIO.cc:235]: Interesting. They do, but when I ran a few tests awhile ago comparing both, I've seen differences in IDs between programmatic and g4 / vecgeom. But loading a gdml file on g4 or vecgeom resulted in equivalent IDs...

**Comment by stognini** [comment]
> @stognini For the sake of completion, could you post the 11.0.0 failures?

Yes. Here's the log:

# Build options

-DCELERITAS_USE_VecGeom=OFF -DCELERITAS_BUILD_DEMOS=ON -DCELERITAS_USE_ROOT=ON -DCELERITAS_USE_Geant4=ON

# Test failure

$ ./celeritas_ext_GeantImporter

```
[==========] Running 10 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 10 tests from FourSteelSlabsEmStandard
[ RUN      ] FourSteelSlabsEmStandard.em_particles
[       OK ] FourSteelSlabsEmStandard.em_particles (115 ms)
[ RUN      ] FourSteelSlabsEmStandard.em_hadronic
[       OK ] FourSteelSlabsEmStandard.em_hadronic (11 ms)
[ RUN      ] FourSteelSlabsEmStandard.elements
[       OK ] FourSteelSlabsEmStandard.elements (10 ms)
[ RUN      ] FourSteelSlabsEmStandard.materials
/Users/4s2/celeritas-project/celeritas/test/celeritas/ext/GeantImporter.test.cc:348: Failure
Values in: cutoff_energies
 Expected: expected_cutoff_energies
3 of 6 elements differ
by 0.02 relative error or 0.00020000000000000001 absolute error
i            EXPECTED    cutoff_energies         Difference
3    1.22808845964606               0.001  -0.999185726409083
4    1.31345289979559               0.001  -0.999238647994035
5  0.0209231725658313               0.001  -0.952206100826552

[  FAILED  ] FourSteelSlabsEmStandard.materials (11 ms)
[ RUN      ] FourSteelSlabsEmStandard.processes
/Users/4s2/celeritas-project/celeritas/test/celeritas/ext/GeantImporter.test.cc:497: Failure
Values in: result.size
 Expected: expected_size
3 of 3 elements differ
i    expected_size      result.size
0                5                6ul
1                5                6ul
2                5                6ul

/Users/4s2/celeritas-project/celeritas/test/celeritas/ext/GeantImporter.test.cc:498: Failure
Values in: result.x_bounds
 Expected: expected_x_bounds
3 of 6 elements differ
by 0.0050000000000000001 relative error or 5.0000000000000002e-05 absolute error
i   expected_x_bounds    result.x_bounds         Difference
0   0.020923172565831               0.001  -0.952206100826551
2   0.020923172565831               0.001  -0.952206100826551
4   0.020923172565831               0.001  -0.952206100826551

/Users/4s2/celeritas-project/celeritas/test/celeritas/ext/GeantImporter.test.cc:499: Failure
Values in: result.y_bounds
 Expected: expected_y_bounds
3 of 6 elements differ
by 0.0050000000000000001 relative error or 5.0000000000000002e-05 absolute error
i   expected_y_bounds    result.y_bounds         Difference
0     19.855602934384    213.526763471911    9.75398033379012
2     16.824420929076    184.768380663338    9.98215394409329
4     23.159721368813    243.834905672938    9.52840411116895

[  FAILED  ] FourSteelSlabsEmStandard.processes (11 ms)
[ RUN      ] FourSteelSlabsEmStandard.volumes
[       OK ] FourSteelSlabsEmStandard.volumes (11 ms)
[ RUN      ] FourSteelSlabsEmStandard.em_parameters
[       OK ] FourSteelSlabsEmStandard.em_parameters (11 ms)
[ RUN      ] FourSteelSlabsEmStandard.sb_data
[       OK ] FourSteelSlabsEmStandard.sb_data (11 ms)
[ RUN      ] FourSteelSlabsEmStandard.livermore_pe_data
[       OK ] FourSteelSlabsEmStandard.livermore_pe_data (11 ms)
[ RUN      ] FourSteelSlabsEmStandard.atomic_relaxation_data
[       OK ] FourSteelSlabsEmStandard.atomic_relaxation_data (11 ms)
[----------] 10 tests from FourSteelSlabsEmStandard (214 ms total)

[----------] Global test environment tear-down
[==========] 10 tests from 1 test suite ran. (214 ms total)
[  PASSED  ] 8 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] FourSteelSlabsEmStandard.materials
[  FAILED  ] FourSteelSlabsEmStandard.processes

 2 FAILED TESTS
```

File: app/demo-loop/LDemoIO.cc
Comment (sethrj) [reply app/demo-loop/LDemoIO.cc:235]: It generally seems to for the simple ones, but they definitely don't match up for CMS2018.

File: app/celer-dump-data.cc
Comment (sethrj) [reply app/celer-dump-data.cc]: I put in a workaround that drops those, yes.

**Review by stognini — APPROVED** [review]

**Comment by sethrj** [comment]
Wow, those cutoff energies I guess are pretty broken for 11.0.0. but fixed for 11.0.2? Thanks for the review!

**Pull request merged**
